packages/renderer/src/lib/image/ImageDetailsFiles.svelte

```diff
@@ -18,48 +18,76 @@ let selectedLayer = $state<ImageFilesystemLayerUI>();
 let loading = $state<boolean>(false);
 let error = $state<string>('');
 let showLayerOnly = $state<boolean>(false);
+let showFetchButton = $state<boolean>(false);
 
 let filesProvidersUnsubscribe: Unsubscriber;
-let filesProvider: ImageFilesInfo;
+let filesProvider: ImageFilesInfo | undefined = undefined;
 let cancellableTokenId: number = 0;
+let askFetchLayers: boolean = true;
 
 function onSelectedLayer(event: CustomEvent<ImageFilesystemLayerUI>) {
   selectedLayer = event.detail;
 }
 
-onMount(async () => {
-  filesProvidersUnsubscribe = imageFilesProviders.subscribe(providers => {
-    if (providers.length === 1 && imageInfo) {
-      filesProvider = providers[0];
-      loading = true;
-      window.getCancellableTokenSource().then(token => {
-        cancellableTokenId = token;
-        window
-          .imageGetFilesystemLayers(filesProvider.id, imageInfo, cancellableTokenId)
-          .then(layers => {
-            imageLayers = layers;
-          })
-          .catch((err: unknown) => {
-            error = String(err);
-          })
-          .finally(() => {
-            loading = false;
-          });
+function fetchImageLayers(provider: ImageFilesInfo, img: ImageInfo) {
+  loading = true;
+  window.getCancellableTokenSource().then(token => {
+    cancellableTokenId = token;
+    window
+      .imageGetFilesystemLayers(provider.id, img, cancellableTokenId)
+      .then(layers => {
+        imageLayers = layers;
+      })
+      .catch((err: unknown) => {
+        error = String(err);
+      })
+      .finally(() => {
+        loading = false;
       });
```

Comment:
wondering why we're using all the then/then/then/ call vs the more readable pattern
try {
  await 1
  await 2
} finally {
  
}

Reply:
yep, I'll change